datastore/datastore/storagebackend/postgresql/postgresql.go

```diff
@@ -331,46 +342,85 @@ func getStringMdataFilter(
 // cleanup performs various cleanup tasks, like removing old observations from the database.
 func cleanup(db *sql.DB) error {
 
+	log.Println("db cleanup started")
+	start := time.Now()
+
 	// start transaction
 	tx, err := db.Begin()
 	if err != nil {
 		return fmt.Errorf("db.Begin() failed: %v", err)
 	}
 	defer tx.Rollback()
 
+	// --- BEGIN define removal functions ----------------------------------
+
+	rmObsOutsideValidRange := func() error {
+
+		loTime, hiTime := common.GetValidTimeRange()
+		cmd := fmt.Sprintf(`
+			DELETE FROM observation
+			WHERE (obstime_instant < to_timestamp(%d)) OR (obstime_instant > to_timestamp(%d))
+		`, loTime.Unix(), hiTime.Unix())
+
+		_, err = tx.Exec(cmd)
+		if err != nil {
+			return fmt.Errorf(
+				"tx.Exec() failed when removing observations outside valid range: %v", err)
+		}
+
+		return nil
+	}
+
+	rmUnrefRows := func(tableName, fkName string) error {
+
+		cmd := fmt.Sprintf(`
+			DELETE FROM %s
+			WHERE id in (
+				SELECT id FROM %s t WHERE NOT EXISTS (
+					SELECT FROM observation obs WHERE %s = t.id
+				)
+			)
+		`, tableName, tableName, fkName)
```

Comment:
Is there a reason not to do use this?
```suggestion
		cmd := fmt.Sprintf(`
			DELETE FROM %s t
			WHERE NOT EXISTS (
				SELECT FROM observation WHERE %s = t.id
			);
		`, tableName, fkName)
```

Reply:
I guess not. Will change ...

Reply:
It worked in some simple testing. Please double-check. :-)

Reply:
Yup, some simple testing on my side shows that it works too! Good job that you spotted that simplification!